File: 6A/6A.cpp
#include<iostream>
#include<algorithm>
using namespace std;


int a[4];

int main()
{
	cin>>a[0]>>a[1]>>a[2]>>a[3];
	sort(a,a+4);
	
	if(a[1]+a[2] > a[3] || a[0]+[1] > a[2])
		cout<<"TRIANGLE";
	else if (a[1]+a[2]==a[3] || a[0]+a[1]==a[2])
		cout<<"SEGMENT";
	else cout<<"IMPOSSIBLE";

	return 0;
}

File: 6A/6A.py
def f(a,b,c):
	if a+b==c or b==c+a or c==a+b:
		return 0
	if a < b+c and b < a+c  and c<a+b:
		return 1
	return -1

a,b,c,d=sorted(map(int,raw_input().split()))
print { 1: "TRIANGLE", 
	0: "SEGMENT",
  	-1: "IMPOSSIBLE" } [max(f(a,b,c),f(b,c,d),f(c,d,a),f(d,a,b))]



